pkg/controller/assignmetadata/assignmetadata_controller.go

```diff
@@ -119,19 +134,25 @@ func (r *AssignMetadataReconciler) Reconcile(request reconcile.Request) (reconci
 		}
 	}
 	deleted = deleted || !assignMetadata.GetDeletionTimestamp().IsZero()
+	tracker := r.tracker.For(gvkAssignMetadata)
 
 	mutator, err := mutation.MutatorForAssignMetadata(assignMetadata)
 	if err != nil {
 		log.Error(err, "Creating mutator for resource failed", "resource", request.NamespacedName)
+		tracker.CancelExpect(assignMetadata)
 	}
 	if !deleted {
 		if err := r.system.Upsert(mutator); err != nil {
 			log.Error(err, "Insert failed", "resource", request.NamespacedName)
+			tracker.CancelExpect(assignMetadata)
+		} else {
+			tracker.Observe(assignMetadata)
```

Comment:
Talking to myself out loud- It's been a minute, I need to remember the nuanced differences between canceling an expectation vs. meeting the expectation.

Reply:
Expect - add a new resource to the list of expected items
CancelExpect -  remove resource from the list, with the expectation not being satisfied
Observe - the resource is observed in the system, and can be removed from the list, with the expectation satisfied

Reply:
I think cancel is probably most relevant WRT constraint templates, where all child resources (constraints) are also stricken from the reconciliation list.

Otherwise it does seem a distinction without a difference.

Reply:
Now that there are retries on error, TryCancel does yield meaningfully different behavior.